homeassistant/components/nest/__init__.py

```diff
@@ -245,3 +256,45 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(DATA_NEST_UNAVAILABLE, None)
 
     return unload_ok
+
+
+class NestEventMediaView(HomeAssistantView):
+    """Returns media for related to events for a specific device.
+
+    This is primarily used to render media for events for MediaSource. The media type
+    depends on the specific device e.g. an image, or a movie clip preview.
+    """
+
+    url = "/api/nest/event_media/{device_id}/{event_id}"
+    name = "api:nest:event_media"
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize NestEventMediaView."""
+        self.hass = hass
+
+    async def get(
+        self, request: web.Request, device_id: str, event_id: str
+    ) -> web.StreamResponse:
+        """Start a GET request."""
+        devices = await get_media_source_devices(self.hass)
+        if not (nest_device := devices.get(device_id)):
+            return self._json_error(
+                f"No Nest Device found for '{device_id}'", HTTPStatus.NOT_FOUND
+            )
+        try:
+            event_media = await nest_device.event_media_manager.get_media(event_id)
+        except GoogleNestException as err:
+            raise HomeAssistantError("Unable to fetch media for event") from err
+        if not event_media:
+            return self._json_error(
+                f"No event found for event_id '{event_id}'", HTTPStatus.NOT_FOUND
+            )
+        media = event_media.media
+        return web.Response(
```

Comment:
If this endpoint is only used for _playing_ media, it would be great if we could use a 301 redirect to the actual content.

Reply:
We chatted about this, and the context here is that there are two steps:
- Convert an image to a URL which must be done within ~30 seconds of the event
- Fetch the bytes from the URL which also must be done within ~30 seconds
- We may also will  transform the bytes in the case of media clips since they are 10 frame mp4s and we want to slow them down a bit

Right now this has a cache of a couple items.  However, we also effectively need to always fetch these images directly when the event happens and persist them to really meet the user expectation of how this should work. So we'll consider a 1gb disk cache or something like this, and I think it may have a similar interface as what is already here.